typescript/infra/scripts/funding/multi-protocol-key-funder.ts

```diff
@@ -0,0 +1,472 @@
+  const supportedChains = config.supportedChainNames;
+  const configuredChains = new Set([
+    ...Object.keys(argv.desiredBalancePerChain),
+    ...Object.keys(argv.desiredKathyBalancePerChain ?? []),
+  ]);
+  const chainsToSkip = new Set(argv.chainSkipOverride);
+
+  validateArguments(
+    chainsToSkip,
+    supportedChains,
+    argv.contextsAndRoles,
+    argv.desiredKathyBalancePerChain,
+  );
+
+  const registry = await config.getRegistry();
+  const chainMetadata = await registry.getMetadata();
+  const chainAddresses = await registry.getAddresses();
+  const chainsToFund = Array.from(supportedChains).filter(
+    (chain) =>
+      configuredChains.has(chain) &&
+      !chainsToSkip.has(chain) &&
+      // only support ethereum and sealevel for now
+      (chainMetadata[chain].protocol === ProtocolType.Ethereum ||
+        chainMetadata[chain].protocol === ProtocolType.Sealevel),
+  );
+
+  // filter chainMetadata to only include chainsToFund
+  const filteredChainMetadata = Object.fromEntries(
+    Object.entries(chainMetadata).filter(([chain]) =>
+      chainsToFund.includes(chain),
+    ),
+  );
+
+  // Add ethereum metadata if we have L2 chains that need it for bridging
+  const needsL1 = chainsToFund.some((chain) => L2_CHAINS.includes(chain));
+
+  if (needsL1) {
+    filteredChainMetadata.ethereum = chainMetadata.ethereum;
+  }
+
+  const multiProtocolProviderOptions = await buildMultiProtocolProviderOptions(
+    environment,
+    needsL1 ? [...chainsToFund, L1_CHAIN] : chainsToFund,
+  );
+
+  const multiProtocolProvider = new MultiProtocolProvider(
+    filteredChainMetadata,
+    multiProtocolProviderOptions,
+  );
+
+  // read address from chain metadata
+  const fundingAddresses: ChainMap<FundingAddresses> = chainsToFund.reduce(
+    (acc, chain) => {
+      acc[chain] = {
+        interchainGasPaymaster:
+          chain === 'lumia'
+            ? '0x9024A3902B542C87a5C4A2b3e15d60B2f087Dc3E'
+            : chainAddresses[chain].interchainGasPaymaster,
+      };
+      return acc;
+    },
+    {} as ChainMap<FundingAddresses>,
+  );
+
+  if (needsL1) {
+    fundingAddresses[L1_CHAIN] = {
+      interchainGasPaymaster: chainAddresses.ethereum.interchainGasPaymaster,
+    };
+  }
+
+  if (chainsToFund.length === 0) {
+    throw new Error('No valid chains to fund');
+  }
+
+  logger.info(
+    { chainsToFund },
+    'Determined chains to fund based on configuration',
+  );
+
+  const multiProtocolContextFunders = Object.entries(argv.contextsAndRoles).map(
+    ([context, roles]) => {
+      const ctx = context as Contexts;
+
+      const fundingConfig: FundingConfig = {
+        skipIgpClaim: argv.skipIgpClaim,
+        fundingThresholdFactor: DEFAULT_FUNDING_THRESHOLD_FACTOR,
+      };
+
+      // add funding discount factor for RC context
+      let fundingDiscountFactor: number | undefined;
+      if (ctx === Contexts.ReleaseCandidate) {
+        fundingDiscountFactor = RC_FUNDING_DISCOUNT_FACTOR;
+      }
```

Comment:
maybe flip it so hyperlane is always normal and !hyperlane context has a discount factor

in case we decide to add neutron it means we don't have to add discount factors for every new context we add